[COPS-758] Error descriptions for secrets manager (#520)
neo4j/docs-status-codes: modules/ROOT/pages/errors/gql-errors/51N7D.adoc

= 51N7D

== Status description
error: system configuration or operation exception - secret provider error. Secret provider `{ <<secretProviderType>> }` returned an error. 

== Explanation

> A call to an upstream secret provider failed with an unexpected error. This could be due to a misconfiguration of the secret provider, network issues, or other problems with the secret provider itself. 

Reviewer: ```suggestion
This error occurs when a call to an upstream secret provider fails with an unexpected error.
This could be due to a misconfiguration of the secret provider, network issues, or other problems with the secret provider itself. 
```

== Example scenarios

> * The DBMS is configured to use Google Cloud Secret Manager, but the DBMS is prevent from making calls to Google Cloud due to a firewall.

Reviewer: ```suggestion
* The DBMS is configured to use Google Cloud Secret Manager, but the DBMS is prevented from making calls to Google Cloud due to a firewall.
```

> * The user attempts to store a secret in Google Cloud Secret Manager, but the secret is larger than Google Cloud Secret Manager supports.

Reviewer: ```suggestion
* The length of the secret that you try to store in Google Cloud Secret Manager is not supported.
```

== Possible solutions

To resolve the issue, examine the logs to determine the underlying cause of the error.

> It may be necessary to resolve it in the underlying provider itself. 

Reviewer: ```suggestion
You may need to resolve it within the underlying provider. 
```

> == Related status codes

Reviewer: Moved it to the Explanation section and added a link. Maybe we can add a sentence about how or in what way these two are related.
```suggestion
```

> 51N7C - Secret provider is not available.

Reviewer: ```suggestion
```
